embulk-docs/src/built-in.rst

```diff
@@ -582,6 +582,226 @@ Options
+
++------------+-----------------------------------------------------------------------------------------------------------------------------+--------------------+
+| option     | description                                                                                                                 | required?          |
++============+=============================================================================================================================+====================+
+| delimiter  | A delimiter character inserted before a suffix number. It must be just 1 non-digit character.                               | ``"_"`` by default |
++------------+-----------------------------------------------------------------------------------------------------------------------------+--------------------+
+| digits     | An integer that specifies the number of zero-filled digits of a suffix number. The suffix number zero-filled to the digits. | optional           |
++------------+-----------------------------------------------------------------------------------------------------------------------------+--------------------+
+| max_length | The length to which the column names are truncated. The column name is truncated before the suffix number.                  | optional           |
++------------+-----------------------------------------------------------------------------------------------------------------------------+--------------------+
+| offset     | An integer where the suffix number starts. The first duplicative column name is suffixed by (```offset``` + 1).             | ``1`` by default   |
++------------+-----------------------------------------------------------------------------------------------------------------------------+--------------------+
+
+.. hint::
+   The procedure to make column names unique is not very trivial. There are many feasible ways. This renaming rule works as follows:
+
+   Basic policies:
+
+   * Suffix numbers are counted per original column name.
+   * Column names are fixed from the first column to the last column.
+
+   Actual procedure applied from the first (leftmost) column to the last (rightmost) column:
+
+   1. Fix the column name as-is with truncating if the truncated name is not duplicated with left columns.
+   2. Suffix the column name otherwise.
+
+      a. Try to append the suffix number for the original column name with truncating.
+      b. Fix it if the suffixed name is not duplicated with left columns nor original columns.
+      c. Retry (a) with the suffix number increased otherwise.
+
+Example
+""""""""
+
+.. code-block:: yaml
+
+    # This configuration suffixes numbers to duplicative column names. (Ex. ["column", "column", "column"] goes to ["column", "column_2", "column_3"].)
+    filters:
+      ...
+      - type: rename
+        rules:
+        - rule: unique_number_suffix
+
 
 Example
 ~~~~~~~~
```

Comment:
I prefer `""""""""` instead of `~~~~~~~~`. How about this?

This existing "Example" is good here as an example for combination with `columns` and `rules` options. But, I think, this is minor case. Because almost users use `rules` only.